text/0309-weak-cross-stack-references.md

```diff
@@ -0,0 +1,194 @@
+# Weak cross-stack references
+
+* **Original Author(s):**: @rix0rrr
+* **Tracking Issue**: #309
+* **API Bar Raiser**: @otaviomacedo
+
+A mechanism to make sure that CDK doesn't prevent resource replacement if there is no impact on downstream stacks.
+
+## Motivation
+
+One of the most magical (and hated) features of the CDK is to transparently transport values from one CloudFormation stack to the other. It makes it possible to pass constructs from one stack and pass them into another, and use the construct as if it was defined locally. CDK will detect that a value originating from one stack is used in a different one, and insert the right CloudFormation incantations to make sure the value is properly referenced.
+
+However, CDK has chosen a mechanism to do this (called **CloudFormation Exports**) which have a specific side effect: *values which are currently being consumed by a different Stack are not allowed to be changed or removed*. This RFC introduces "weak cross-stack references", a way to make certain cross-stack references selectively get moved via SSM Parameters instead of by CloudFormation Exports.
+
+We introduce a mechanism for turning strong references into weak references, and turn this on by default for CloudWatch Dashboards.
```

Comment:
>turn this on by default for CloudWatch Dashboards.

Default only for it, am I reading this correctly?

Reply:
Yes, explained as an use-case in a subsequent section.

Reply:
Yes.